lib/md5.c

```diff
@@ -90,6 +90,34 @@ static void MD5_Final(unsigned char digest[16], MD5_CTX *ctx)
 /* The last #include file should be: */
 #include "memdebug.h"
 
+#elif defined(USE_MBEDTLS)
+
+#include <mbedtls/md5.h>
+
+#include "curl_memory.h"
+
+/* The last #include file should be: */
+#include "memdebug.h"
+
+typedef mbedtls_md5_context MD5_CTX;
+
+static void MD5_Init(MD5_CTX *ctx)
+{
+  (void) mbedtls_md5_starts_ret(ctx);
+}
+
+static void MD5_Update(MD5_CTX *ctx,
+                       const unsigned char *data,
+                       unsigned int length)
+{
+  (void) mbedtls_md5_update_ret(ctx, data, length);
+}
+
+static void MD5_Final(unsigned char digest[16], MD5_CTX *ctx)
```

Comment:
I get why you did this (because mbedtls does I assume) but I think the fixed size array style is misleading, if you sizeof(digest) it's going to be the size of the pointer not of the array. If we are passing *input I think it should just be passed as that.

Reply:
NEvermind it looks like that's the current convention, I'd stick with it.

Reply:
Unfortunately it's not related to mbedtls. All of our static `MD5_` functions use this style and you'll also note that the SHA256 code does this too (I'm guessing as the MD5 code was used as a template).

I'm personally not a fan if it because of it for the reason you gave ;-) As such you'll note I've used pointer notation in PR #4956 for my new SHA256 functions.